src/detect-engine-mpm.c

```diff
@@ -772,18 +772,12 @@ int SignatureHasPacketContent(const Signature *s)
 {
     SCEnter();
 
-    if (s == NULL) {
-        SCReturnInt(0);
-    }
-
     if (!(s->proto.proto[IPPROTO_TCP / 8] & 1 << (IPPROTO_TCP % 8))) {
         SCReturnInt(1);
     }
 
-    if ((s->init_data != NULL && s->init_data->smlists[DETECT_SM_LIST_PMATCH] == NULL) ||
-        (s->init_data == NULL && s->sm_arrays[DETECT_SM_LIST_PMATCH] == NULL))
```

Comment:
Have you deleted the check for `s->sm_arrays[DETECT_SM_LIST_PMATCH] == NULL` because `s->init_data == NULL`  can't happen?

Reply:
yes, we can't get here (anymore) if s->init_data == NULL